pvacseq/lib/parse_output.py

```diff
@@ -15,63 +15,186 @@
 
 csv.field_size_limit(sys.maxsize)
 
+def parse_input_tsv_file(input_tsv_file):
+    tsv_reader = csv.DictReader(input_tsv_file, delimiter='\t')
+    tsv_entries = {}
+    for line in tsv_reader:
+        tsv_entries[line['index']] = line
+    return tsv_entries
+
 def min_match_count(peptide_length):
     return ceil(peptide_length / 2)
 
-def determine_consecutive_matches(mt_epitope_seq, wt_epitope_seq):
+def determine_consecutive_matches_from_left(mt_epitope_seq, wt_epitope_seq):
     consecutive_matches = 0
-    left_padding        = 0
-    #Count consecutive matches from the beginning of the epitope sequences
     for a, b in zip(mt_epitope_seq, wt_epitope_seq):
         if a == b:
             consecutive_matches += 1
-            left_padding        += 1
         else:
             break
-    #Count consecutive matches from the end of the epitope sequences
+    return consecutive_matches
+
+def determine_consecutive_matches_from_right(mt_epitope_seq, wt_epitope_seq):
+    consecutive_matches = 0
     for a, b in zip(reversed(mt_epitope_seq), reversed(wt_epitope_seq)):
         if a == b:
             consecutive_matches += 1
         else:
             break
-    return consecutive_matches, left_padding
-
-def parse_input_tsv_file(input_tsv_file):
-    tsv_reader = csv.DictReader(input_tsv_file, delimiter='\t')
-    tsv_entries = {}
-    for line in tsv_reader:
-        tsv_entries[line['index']] = line
-    return tsv_entries
+    return consecutive_matches
+
+def find_mutation_position_from_left(wt_epitope_seq, mt_epitope_seq):
+    for i,(wt_aa,mt_aa) in enumerate(zip(wt_epitope_seq,mt_epitope_seq)):
+        if wt_aa != mt_aa:
+            return i+1
+    return 0
+
+def find_mutation_position_for_insertions(wt_epitope_seq, mt_epitope_seq):
+    while len(wt_epitope_seq) > 0:
+        wt_epitope_seq = wt_epitope_seq[1:]
+        mt_epitope_seq = mt_epitope_seq[:-1]
+        for i,(wt_aa,mt_aa) in enumerate(zip(wt_epitope_seq,mt_epitope_seq)):
+            if wt_aa != mt_aa:
+                mutation_position = i
+                break
+        if mutation_position != 0:
```

Comment:
Good question. Now that you pointed it out I realized that this whole subroutine isn't doing the right thing at all. I pushed a commit to fix this bug.